### PR TITLE
#166789874 Add user resource to view or update profile

### PR DESCRIPTION
**Pull request by adafia**
#### What does this PR do?
This pull request adds resources that allow users to: 
 - view their profiles as well as any other users' profile.
 - update only their profiles

#### Description of Task to be completed?
Users can view their profiles as well as that of other users but can only edit and update their profile.
> the feature can be accessed and tested via `https://ah-lobos-backend-swagger.herokuapp.com/`

#### How should this be manually tested?
Follow instructions on the `README.md` file attached to this repo to set up the project on your local machine.
Access the following routes to test
> User view profile `api/profile/:username`
> User update profile `api/profiles/:username`

#### Any background context you want to provide?
Profiles will provide a way to maintain generic and user-specific data/info such as username, bio, contact info, etc.

#### What are the relevant pivotal tracker stories?
[#166789874](https://www.pivotaltracker.com/story/show/166789874)

#### Screenshots (if appropriate)
Swagger documentation
Screen Shot 2019-07-09 at 17 22 55

- User view profile
Screen Shot 2019-07-09 at 20 12 27
Screen Shot 2019-07-09 at 20 12 49

- user update profile
Screen Shot 2019-07-10 at 11 18 21

Screen Shot 2019-07-10 at 11 18 45

-Mockup Screenshots
update profile
Screen Shot 2019-07-11 at 11 39 06

View user profile 
Screen Shot 2019-07-11 at 12 32 21

#### Questions:
N/A

### CODE REVIEW

**Review by manzif — APPROVED** [review]

**Review by Dom58 — COMMENTED** [review]
In PR description can you insert the `Heroku ` url documentation to review through it ourself !

**Review by j4l13n — APPROVED** [review]

**Review by Emile-Nsengimana — APPROVED** [review]

**Comment by adafia** [comment]
> In PR description can you insert the `Heroku ` url documentation to review through it ourself !

**Review by emukungu — CHANGES_REQUESTED** [review]
@adafia, The files are too long. Separate the profile management from authentication. 
Share your mockups on the PR under screen shots.

**Comment by adafia** [comment]
> @adafia, The files are too long. Separate the profile management from authentication.
> Share your mockups on the PR under screenshots.

I'll work on separating them as soon as possible. I'm currently working on the mockups, I'll update it as soon as I'm done.

File: tests/userProfileTest.js [hunk 1]
```diff
@@ -0,0 +1,173 @@
+import jwt from 'jsonwebtoken';
+import dotenv from 'dotenv';
+import Chai from 'chai';
```
Comment (emukungu) [inline tests/userProfileTest.js]: change  'Chai' to lowercase

File: tests/userProfileTest.js [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+const payload1 = {
+  username: 'Annor',
+  email: 'annor@andela.com'
+};
+
+const token = jwt.sign(payload, process.env.SECRET_JWT_KEY, { expiresIn: '24h' });
+const token1 = jwt.sign(payload1, process.env.SECRET_JWT_KEY, { expiresIn: '24h' });
+
+describe('User Profile', () => {
+  before('Before testing user view/update profile, sign-up user one', (done) => {
```
Comment (emukungu) [inline tests/userProfileTest.js]: why do your tests have very many before statements, Group them into one.

File: tests/userProfileTest.js (hunk 1 above)
Comment (adafia) [reply tests/userProfileTest.js]: Noted, i'll change it as soon as possible

File: tests/userProfileTest.js (hunk 2 above)
Comment (adafia) [reply tests/userProfileTest.js]: Okay

**Comment by adafia** [comment]
> @adafia, The files are too long. Separate the profile management from authentication.
> Share your mockups on the PR under screenshots.

Done. Please check them out

**Review by Dom58 — APPROVED** [review]

**Pull request merged**